File: springmvc/spring-springmvc-springdatajpa/src/main/java/com/jiangKlijna/web/control/UserControl.java
package com.jiangKlijna.web.control;

import javax.annotation.Resource;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import com.jiangKlijna.web.bean.Result;
import com.jiangKlijna.web.service.UserService;
import org.springframework.web.bind.annotation.ResponseBody;

@RequestMapping("/user")
@Controller
public class UserControl extends BaseControl {

    @Resource(name = "userService")
    public UserService us;

    @ResponseBody
    @RequestMapping("/regist.json")
    public Result regist_json(String username, String password) {
        return testParameter(username, password) ? us.regist(username, password) : errorParameterResult;
    }

    @ResponseBody
    @RequestMapping("/regist.xml")
    public Result regist_xml(String username, String password) {
        return testParameter(username, password) ? us.regist(username, password) : errorParameterResult;
    }

    @ResponseBody
    @RequestMapping("/remove.json")
    public Result remove(int id) {
        return us.remove(id);
    }

    @ResponseBody
    @RequestMapping("/find.xml")
    public Result find(int id) {
        return us.find(id);
    }
}

File: springmvc/spring-springmvc-mybatis-kotlin/src/main/java/com/jiangKlijna/web/control/UserControl.kt
package com.jiangKlijna.web.control

import com.jiangKlijna.web.bean.Result
import com.jiangKlijna.web.service.UserService
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody

import javax.annotation.Resource

@RequestMapping("/user")
@Controller
class UserControl : BaseControl() {

    @Resource(name = "userService")
    var us: UserService? = null

    @ResponseBody
    @RequestMapping("/regist.json")
    fun regist_json(username: String, password: String): Result {
        return if (BaseControl.testParameter(username, password)) us!!.regist(username, password) else errorParameterResult
    }

    @ResponseBody
    @RequestMapping("/regist.xml")
    fun regist_xml(username: String, password: String): Result {
        return if (BaseControl.testParameter(username, password)) us!!.regist(username, password) else errorParameterResult
    }

    @ResponseBody
    @RequestMapping("/remove.json")
    fun remove(id: Int): Result {
        return us!!.remove(id)
    }

    @ResponseBody
    @RequestMapping("/find.xml")
    fun find(id: Int): Result {
        return us!!.find(id)
    }
}
